common/src/dodeca.rs

```diff
@@ -361,25 +402,51 @@ lazy_static! {
         }
         assert_eq!(vertex.next(), None);
         result
-    };
+    })
+}
 
-    /// Whether the determinant of the cube-to-node transform is negative
-    static ref CHUNK_TO_NODE_PARITY: [bool; VERTEX_COUNT] = {
+/// Whether the determinant of the cube-to-node transform is negative
+fn chunk_to_node_parity_static() -> &'static [bool; VERTEX_COUNT] {
+    static LOCK: OnceLock<[bool; VERTEX_COUNT]> = OnceLock::new();
+    LOCK.get_or_init(|| {
         let mut result = [false; VERTEX_COUNT];
 
         for v in Vertex::iter() {
             result[v as usize] = math::parity(&v.chunk_to_node_f64());
         }
 
         result
-    };
-
-    static ref SIDE_NORMALS_F32: [na::Vector4<f32>; SIDE_COUNT] = SIDE_NORMALS_F64.map(|n| n.cast());
-    static ref REFLECTIONS_F32: [na::Matrix4<f32>; SIDE_COUNT] = REFLECTIONS_F64.map(|n| n.cast());
-    static ref DUAL_TO_NODE_F32: [na::Matrix4<f32>; VERTEX_COUNT] = DUAL_TO_NODE_F64.map(|n| n.cast());
-    static ref NODE_TO_DUAL_F32: [na::Matrix4<f32>; VERTEX_COUNT] = NODE_TO_DUAL_F64.map(|n| n.cast());
-    static ref DUAL_TO_CHUNK_FACTOR_F32: f32 = *DUAL_TO_CHUNK_FACTOR_F64 as f32;
-    static ref CHUNK_TO_DUAL_FACTOR_F32: f32 = *CHUNK_TO_DUAL_FACTOR_F64 as f32;
+    })
+}
+
+fn side_normals_f32_static() -> &'static [na::Vector4<f32>; SIDE_COUNT] {
+    static LOCK: OnceLock<[na::Vector4<f32>; SIDE_COUNT]> = OnceLock::new();
+    LOCK.get_or_init(|| side_normals_f64_static().map(|n| n.cast()))
+}
+
+fn reflections_f32_static() -> &'static [na::Matrix4<f32>; SIDE_COUNT] {
+    static LOCK: OnceLock<[na::Matrix4<f32>; SIDE_COUNT]> = OnceLock::new();
+    LOCK.get_or_init(|| reflections_f64_static().map(|n| n.cast()))
+}
+
+fn dual_to_node_f32_static() -> &'static [na::Matrix4<f32>; VERTEX_COUNT] {
+    static LOCK: OnceLock<[na::Matrix4<f32>; VERTEX_COUNT]> = OnceLock::new();
+    LOCK.get_or_init(|| dual_to_node_f64_static().map(|n| n.cast()))
+}
+
+fn node_to_dual_f32_static() -> &'static [na::Matrix4<f32>; VERTEX_COUNT] {
+    static LOCK: OnceLock<[na::Matrix4<f32>; VERTEX_COUNT]> = OnceLock::new();
+    LOCK.get_or_init(|| node_to_dual_f64_static().map(|n| n.cast()))
+}
+
+fn dual_to_chunk_factor_f32_static() -> &'static f32 {
```

Comment:
Don't return (or pass) floats by reference. The reference is bigger then the bare float would be.